### PR TITLE
Allow support for mysql5

### PR DESCRIPTION
**Pull request by caryssaperez**
So this was the impetus for me wanting to contribute: 

We use Accessly at Iris, because it's great, and we had some issues with the UUID support. Unfortunately, we're plebes and still use MySQL 5 which does not support UUIDs out the box without some finagling (like doing something cool like [this](https://stackoverflow.com/questions/43222114/rails-5-mysql-uuid)). However, the format that Accessly's migrations are in use

```ruby
 create_table :accessly_permitted_action_on_objects, force: true, id: :uuid
```

which is syntax that our DB still interprets as "Make the ID an integer". Understandably, 

```ruby
Accessly::PermittedAction.create!(
  id: SecureRandom.uuid,
  segment_id: -1,
  actor: user,
  action: 1
)
```

yields some _interesting_ results for our IDs. Since we don't have granular control of the syntax for the migrations, and we need something implemented rather quickly, I wanted to make a change to how IDs are set in a `before_create` hook on our two lovely models.

The hook just checks the type of the ID column that's actually saved in the table. If it's a UUID, then we save a UUID. If it's not, then the DB will do the rest since primary key columns are usually integers set to auto-increment anyway. (I could have checked for `ActiveRecord::Base.connection.adapter_name` or checked for an instance of the Postgres adapter, but I don't want to narrow it to just Postgres).

Lemme know if this is alright with y'all. Totally open to any tweaks or suggestions.

### CODE REVIEW

**Comment by rreinhardt9** [comment]
@caryssaperez thanks for looking into UUID support! That does seem like something that would be helpful to figure out how to handle appropriately.

I'll have to dig in a bit to understand this PR and the possible approaches here and will probably ping some other maintainers from this PR to get their input on this one. But just wanted ya to know, it's on my radar now 🎉

File: lib/accessly/models/permitted_action.rb
Comment (caryssaperez) [inline lib/accessly/models/permitted_action.rb:12]:
```suggestion
      return unless PermittedAction.columns_hash['id'].type == :uuid
```

**Comment by rreinhardt9** [comment]
Hi again @caryssaperez !!! Sorry, it's taken me a little bit to wrap my mind around this 😬 I've got a few follow up questions to make sure I'm tracking right on this.

I think I understood this backwards before... I was thinking you were trying to add support for UUID in mysql. If I'm understanding correctly what this would actually allow us to do is NOT use UUIDs in mysql. The goal is to only try and set the UUID IF the column type is of the type `UUID`? The end result for mysql should be that it just lets the normal integer based primary key increment when new records are created. Am I understanding that correctly?

One part I've been confused about is why we were setting the UUID manually in Accessly to begin with 🤔.  I think if we didn't explicitly set the UUID that postgres would automatically generate the UUID when creating the new record. If that is the case, then we might be able to just remove all the places that we were setting UUID explicitly (like you already have) without needing to add the `before_create` hook.

If you were to remove the before_create hooks in this PR and run the specs... do they still pass? I'm wondering if we could make it work both for mysql and postgresql in that way....

**Comment by caryssaperez** [comment]
@rreinhardt9 Yes exactly! And lemme actually try removing the hook to see what happens. Theoretically, the DB should just take over. Will respond in a sec.

**Comment by caryssaperez** [comment]
Yep still passes! I'll go ahead and remove the hook.

**Comment by caryssaperez** [comment]
Done!

**Comment by rreinhardt9** [comment]
Awesome @caryssaperez !!! Wow, I'm really wondering why we had the code in there explicitly setting it before then 🤔 I'll poke around with this a bit locally and then ask for feedback with some others on the Lessonly team. Thanks for working on this, it will really clean that up nicely and enhance it's usability if we can make this change 🎉

**Comment by rreinhardt9** [comment]
@caryssaperez I've been doing some local testing in a few apps to make sure this is still working as expected (I'm really nervous about the fact that I can't put a finger on why it was there in the first place 😆 ). So far things are looking just fine, it seems to work as expected!

If you get a chance would you be able to resolve the conflicts from merging the test update branch? The final thing I'm going to try to do is to try to get someone else from our team to give me a sanity check, but it looks like this will work for us so far! 🎉

**Comment by caryssaperez** [comment]
Already did @rreinhardt9 !

File: test/revoke_general_action_test.rb
Comment (rreinhardt9) [inline test/revoke_general_action_test.rb:8]: When I run the specs now, I'm getting those deprecation warnings from this that you fixed for us (:tada:) in that other PR. Would we be able to change this back to the `_()` syntax to silence those warnings again? The test output is so nice now thanks to that other PR!

**Review by rreinhardt9 — CHANGES_REQUESTED** [review]
Hey @caryssaperez !!! I've been asking around and digging into this more myself, and I think we are clear for this change!!

The conclusion after asking around is that there is really no benefit that we can think of (performance or otherwise) for setting the UUID explicitly in the Ruby code during create instead of relying on the DB function. The only possible advantage we thought of was so that you could know the ID of the record before it was created and without having to retrieve it from the DB again. So I think we're good to merge this after we update those tests to the new syntax!

Would you mind squashing the commits down to one with some context on this change after the test are updated? Then I think we'll be good to go 🚀 Thanks for your insight into this and helping make it more broadly usable by people!

### MySQL side-notes from conversations
Just a few side notes from my co-workers on MySQL and UUID that I thought I'd pass along (might already be stuff you've covered, but just in case it's useful):

One of them mentioned in passing that he thought you could store UUIDs in a "binary" type column in mysql if desired. 

He wasn't sure how that would work with the Active Record MySQL adapter, but send me this link for a gem that looked like it added support for UUID: https://github.com/mathieujobin/activerecord-mysql-uuid-column

Those notes were just separate from this PR; this change seems valuable apart from any of the mysql specific UUID work.

File: test/revoke_general_action_test.rb
Comment (caryssaperez) [inline test/revoke_general_action_test.rb:8]:
```suggestion
    _(Accessly::PermittedAction.where(actor: actor).count).must_equal 1
```

File: test/revoke_general_action_test.rb
Comment (caryssaperez) [inline test/revoke_general_action_test.rb:17]:
```suggestion
    _(Accessly::PermittedAction.where(actor: actor, segment_id: 1).count).must_equal 1
```

File: test/revoke_records_test.rb
Comment (caryssaperez) [inline test/revoke_records_test.rb:10]:
```suggestion
    _(Accessly::PermittedActionOnObject.where(actor: actor).count).must_equal 1
```

File: test/revoke_records_test.rb
Comment (caryssaperez) [inline test/revoke_records_test.rb:21]:
```suggestion
    _(Accessly::PermittedActionOnObject.where(actor: actor, segment_id: 1).count).must_equal 1
```

**Comment by caryssaperez** [comment]
@rreinhardt9 Sorry for the delay, had a bunch of shit to handle at work. I updated the specs.

**Comment by rreinhardt9** [comment]
So awesome, thanks again for this @caryssaperez 🎉 !!! Let's :shipit:

**Pull request merged**
